nbconvert/preprocessors/clearmetadata.py

```diff
@@ -0,0 +1,22 @@
+"""Module containing a preprocessor that removes metadata from code cells"""
+
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from traitlets import Set
+from .base import Preprocessor
+
+class ClearMetadataPreprocessor(Preprocessor):
+    """
+    Removes the output from all code cells in a notebook.
+    """
+
+    def preprocess_cell(self, cell, resources, cell_index):
+        """
+        Apply a transformation on each cell. See base.py for details.
+        """
+        if cell.cell_type == 'code':
+            # Remove metadata
+            if 'metadata' in cell:
+                cell.metadata = {}
```

Comment:
@takluyver we've had lots of issues about this — I actually think this is going to be one of the cleanest ways to implement this.

Reply:
Version control was the purpose of develop this. With this feature it will be possible to apply a clean and smudge for notebooks.